File: RegexProjects/hmwk_02.cc
// Basnet, Himal
// hxb9428
// 2020-04-01

#include <fstream>
#include <iostream>
#include <regex>

using namespace std;

//----------------------------------------------------------
regex r1("^(%|\\*)(([%][%])*|([\\*][\\*])*|([\\*][%])*|([%][\\*])*)*([pJ]+|[p]+(?:pay)|[J]+(?:JAY))$");
//regex r2("^(\\+|-|:)([A-F0-9][A-F0-9])+(\\W(?!\1))$");
regex r2("^(\\+([0-9A-F][0-9A-F])*(-|:)|-([0-9A-F][0-9A-F])*(\\+|:)|:([0-9A-F][0-9A-F])*(-|\\+))$");
regex r3("^\\|([AEIOU]+(?:\\*)|[aeiou]+(?:=)|\\$)$");

smatch m1,m2,m3;

void processToken( string token )
{
  if(regex_match(token, r1))
    cout << ">"+token+"< matches PayJay." << endl;
  else if(regex_match(token, r2))
    cout << ">"+token+"< matches Rattle." << endl;
  else if(regex_match(token, r3))
    cout << ">"+token+"< matches Tork." << endl;
  else
    cout << ">"+token+"< does not match." << endl;
}

//----------------------------------------------------------
// DO NOT CHANGE anything below this line!
int main( int argc, char *argv[] )
{
  if ( argc > 1 ) {
    cout << "processing tokens from " << argv[ 1 ] << " ...\n";

    ifstream inputFile;
    string   token;

    inputFile.open( argv[ 1 ] );

    if ( inputFile.is_open() ) {
      while ( inputFile >> token ) {
        processToken( token );
      }
      inputFile.close();
    } else {
      cout << "unable to open " << argv[ 1 ] << "?\n";
    }
  } else {
    cout << "No input file specified.\n";
  }
}

//----------------------------------------------------------

File: RegexProjects/hmwk_02.java
// Basnet, Himal
// hxb9428
// 2020-04-01

import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.Arrays;

//----------------------------------------------------------
public class hmwk_02 {
  public static void processToken(String token) {
    if (token.matches("^(%|\\*)(([%][%])*|([\\*][\\*])*|([\\*][%])*|([%][\\*])*)*([pJ]+|[p]+(?:pay)|[J]+(?:JAY))$")) {
      System.out.println(">" + token + "< matches PayJay.");
    } else if (token.matches("^(\\+([0-9A-F][0-9A-F])*(-|:)|-([0-9A-F][0-9A-F])*(\\+|:)|:([0-9A-F][0-9A-F])*(-|\\+))$")) {
      System.out.println(">" + token + "< matches Rattle.");
    } else if (token.matches("^\\|([AEIOU]+(?:\\*)|[aeiou]+(?:=)|\\$)$")) {
      System.out.println(">" + token + "< matches Tork.");
    } else {
      System.out.println(">" + token + "< does not match.");
    }
  }

  // --------------------------------------------------------
  // DO NOT CHANGE anything below this line!
  public static void main(String[] args) {
    System.out.println("processing tokens from " + args[0] + " ...");

    try {
      Files.lines(Paths.get(args[0])).filter(line -> line.length() > 0)
          .forEachOrdered(line -> Arrays.stream(line.split("\\s+")).forEachOrdered(token -> processToken(token)));
    } catch (java.io.IOException e) {
      e.printStackTrace();
    }
  }
}

// ----------------------------------------------------------
